src/NextGen/identity/identity.ts

```diff
@@ -0,0 +1,31 @@
+import { MPID, User } from '@mparticle/web-sdk';
+import { IMParticleComponent } from '../component';
+import { IdentityCallback } from './identityCallback';
+
+// QUESTION: What is the diff between this and IdentityApiData?
+// TODO: sync this with android class
+export interface IdentityApiResult {}
+export interface IdentityApiRequest {}
+
+// QUESTION: Should we mirror the way Android handles HTTP codes, or do it the way we've been doing it?
+// https://docs.mparticle.com/developers/sdk/android/idsync/#idsync-status-codes
+// https://docs.mparticle.com/developers/sdk/web/idsync/#error-handling
+
+export interface IMparticleIdentity extends IMParticleComponent {
+    // QUESTION: Should these be callbacks or promises that eventually return something?
```

Comment:
I'd argue we should move as much to promises as possible.

Reply:
Promises is a new concept to me (thought it seems like its a thing in Swift though it kinda still looks like a handler to me https://www.swiftbysundell.com/articles/under-the-hood-of-futures-and-promises-in-swift/ )

Reply:
Yes everything transpiles.  We just have to ensure we have the proper test coverage for browsers when it comes to the transpiling.

Reply:
Yep no need for callbacks then at all on web